pandas/core/arrays/period.py

```diff
@@ -82,7 +94,186 @@ def wrapper(self, other):
     return compat.set_function_name(wrapper, opname, cls)
 
 
-class PeriodArrayMixin(DatetimeLikeArrayMixin):
+class PeriodArray(DatetimeLikeArrayMixin, ExtensionArray):
+    """
+    Pandas ExtensionArray for Period data.
+
+    There are two components to a PeriodArray
+
+    - ordinals
+    - freq
+
+    The values are physically stored as an ndarray of integers. These are
+    called "ordinals" and represent some kind of offset from a base.
+
+    The `freq` indicates the span covered by each element of the array.
+    All elements in the PeriodArray have the same `freq`.
+    """
+    _attributes = ["freq"]
+    # --------------------------------------------------------------------
+    # Constructors
+
+    def __new__(cls, data=None, ordinal=None, freq=None, start=None, end=None,
+                periods=None, tz=None, dtype=None, copy=False,
```

Comment:
tz very likely doesnt belong

Reply:
Part of why the constructors for the existing mixins are bare-bones is because there are comments in the Index subclasses suggesting things like start/end should be taken out of them.

Reply:
And I think we could then combine `_simple_new` and `_from _ordinals` ?

Reply:
No preference on doing it as part of this PR or another. The diff is already impossible to read, so I may as well try doing it here.